src/WebJobs.Extensions.DurableTask/Listener/TaskOrchestrationShim.cs

```diff
@@ -117,7 +124,60 @@ public void SetFunctionInvocationCallback(Func<Task> callback)
 
             if (returnValue != null)
             {
-                this.context.SetOutput(returnValue);
+                if (orchestratorInfo.IsOutOfProc && returnValue is JObject)
+                {
+                    var jObj = returnValue as JObject;
+                    if (jObj["actions"] != null)
+                    {
+                        var commandSets = JsonConvert.DeserializeObject<AsyncCommand[][]>(jObj["actions"].ToString());
```

Comment:
I see a few places where we're doing dictionary lookups into `jObj`. Would it make sense to just deserialize the entire `jObj` into a strongly-typed class?  I see we're doing that for the `AsyncCommand[][]` but I'm wondering if we should deserialize starting with the outer `jObj`.